backend/open_api_spec.yml

```diff
@@ -0,0 +1,683 @@
+        id:
+          type: string
+        dataset_id:
+          type: string
+        filetype:
+          type: string
+          enum: [H5AD, RDS, CXG]
+        filename:
+          type: string
+        type:
+          type: string
+          enum: [REMIX, ORIGINAL]
+        s3_uri:
+          type: string
+    gene_set:
+      type: object
+      properties:
+        name:
+          type: string
+        id":
+          type: string
+        description:
+          type: string
+        linked_datasets:
+          type: array
+          items:
+            type: string
+    ontology_element:
+      type: object
+      properties:
+        ontology_term_id:
+          type: string
+        label:
+          type: string
+    ontology_element_array:
+      type: array
+      items:
+        $ref: "#/components/schemas/ontology_element"
+    problem:
+      type: object
+      description: Error message container for HTTP APIs.
+      properties:
+        type:
+          type: string
+        title:
+          type: string
+        detail:
+          type: string
+    collection_form_metadata:
+      type: object
+      additionalProperties: false
+      required:
+        - name
+        - description
+        - contact_name
+        - contact_email
+        - curator_name
+      properties:
+        name:
+          type: string
+          description: name of the collection
+        description:
+          type: string
+          description: description of the collection
+        contact_name:
+          type: string
+          description: name of the primary person of contact for the collection
+        contact_email:
+          type: string
+          description: email of contact person for the collection
+        curator_name:
+          type: string
+          description: name of the curator of the collection.
+        links:
+          $ref: "#/components/schemas/links"
+    upload_link:
+      description: A user-provided link to the dataset source file.
+      type: string
+
+  parameters:
+    optional_query_curator_tag:
+      in: query
+      required: false
+      schema:
+        $ref: "#/components/schemas/curator_tag"
+    optional_query_dataset_uuid:
+      in: query
+      required: false
+      schema:
+        $ref: "#/components/schemas/dataset_uuid"
+    path_collection_uuid:
+      name: collection_uuid
+      description: >
+        The uuid of a collection. To specify the private revision of a public
+        collection, use the uuid of the public collection.
+      in: path
+      required: true
+      schema:
+        $ref: "#/components/schemas/collection_uuid"
+    query_user_uuid:
+      name: user_uuid
+      description: an ID that represents the user.
+      in: query
+      schema:
+        $ref: "#/components/schemas/user_uuid"
+    query_visibility:
+      name: visibility
+      in: query
+      schema:
+        $ref: "#/components/schemas/visibility"
+        default: PUBLIC
+
+  responses:
```

Comment:
Seems a bit weird that we should provide descriptions here, as these are standard HTTP codes.